src/oauth2/init.js

```diff
@@ -1,34 +1,41 @@
-var OicCliError = require('../exception').OicCliError;
+var CLIENT_AUTHN_METHOD = require('../clientAuth').CLIENT_AUTHN_METHOD;
+var ClientInfo = require('../clientInfo.js');
+var Factory = require('../oauth2/service').Factory;
 var HTTPLib = require('../http').HTTPLib;
 var KeyJar = require('../../oicMsg/src/models/keystore-dependency/KeyJar');
-var ClientInfo = require('../clientInfo.js');
-var CLIENT_AUTHN_METHOD = require('../clientAuth').CLIENT_AUTHN_METHOD;
+var OicCliError = require('../exception').OicCliError;
+var Service = require('../service');
 var service = require('../oauth2/service');
 
-var DEFAULT_SERVICES = ['Authorization', 'AccessToken',
-                    'RefreshAccessToken', 'ProviderInfoDiscovery'];
+var DEFAULT_SERVICES = [
+  'Authorization', 'AccessToken', 'RefreshAccessToken', 'ProviderInfoDiscovery'
+];
 
 ExpiredToken.prototype = new OicCliError();
```

Comment:
Could you use a library to use inheritance? Or move the logic to a separate file as it's used in different parts.

Reply:
Fixed